test/overseer/core_test.clj

```diff
@@ -21,125 +14,5 @@
-        "It excludes jobs that are :aborted")))
-
-(deftest test-transitive-dependents
-  (let [conn (test-utils/connect)
-        graph [{:db/id (d/tempid :db.part/user -1001)
-                :job/id "12345"}
-               {:db/id (d/tempid :db.part/user -1002)
-                :job/dep (d/tempid :db.part/user -1001)}
-               {:db/id (d/tempid :db.part/user -1003)
-                :job/dep (d/tempid :db.part/user -1001)}
-               {:db/id (d/tempid :db.part/user -1004)
-                :job/dep (d/tempid :db.part/user -1002)}
-               {:db/id (d/tempid :db.part/user -1005)
-                :job/id "67890"}]
-        {:keys [tempids db-after] :as txn} @(d/transact conn graph)
-        resolve-tempid (fn [tempid]
-                         (->> (d/tempid :db.part/user tempid)
-                              (d/resolve-tempid db-after tempids)) )
-        entity-ids (map resolve-tempid [-1002 -1003 -1004])
-        ->job-id #(:job/id (d/entity db-after %))]
-    (is (= (set (map ->job-id entity-ids))
-           (core/transitive-dependents db-after (->job-id (resolve-tempid -1001)))))
-    (is (= #{}
-           (core/transitive-dependents db-after (->job-id (resolve-tempid -1005)))))))
-
-(deftest test-job-dep-edges
-  (let [graph
-        {:start []
-         :result1 [:start]
-         :result2 [:start]
-         :publish [:result1 :result2]}
-        jobs-by-type (core/job-txns-by-type (keys graph) {})
-        ->job-tempid (fn [job-type] (:db/id (get jobs-by-type job-type)))
-        edge-txns (core/job-dep-edges graph jobs-by-type)]
-      (is (= (->job-tempid :start)
-             (:job/dep (nth edge-txns 0)))
-          ":result1 depends on :start")
-
-      (is (= (->job-tempid :start)
-             (:job/dep (nth edge-txns 1)))
-          ":result2 depends on :start")
-
-      (is (= (->job-tempid :result1)
-             (:job/dep (nth edge-txns 2)))
-          ":publish depends on :result1")
-
-      (is (= (->job-tempid :result2)
-             (:job/dep (nth edge-txns 3)))
-          ":publish depends on :result2")))
-
-(deftest test-status-txn
-  (testing "with no failure"
-    (let [job-id "12345"
-          status :finished
-          job {:overseer/status status}
-          output-txn (core/status-txn job job-id)]
-      (is (= [:job/id job-id] (:db/id output-txn)))
-      (is (= status (:job/status output-txn)))
-      (is (= false (contains? output-txn :job/failure)))))
-  (testing "with failure"
-    (let [job-id "12345"
-          status :failed
-          failure {:foo :bar}
-          job {:overseer/status status
-               :overseer/failure failure}
-          output-txn (core/status-txn job job-id)]
-      (is (= [:job/id job-id] (:db/id output-txn)))
-      (is (= status (:job/status output-txn)))
-      (is (= (pr-str failure) (:job/failure output-txn))))))
+    (is (= #{} (core/missing-handlers handlers g1)))
+    (is (= #{:quux} (core/missing-handlers handlers g2)))))
```

Comment:
Where to put a generic test suite that exercises all aspects of the protocol and you just pass it implementations and verifies the implementation matches the contract?